include/networkit/algebraic/CSRGeneralMatrix.hpp

```diff
@@ -395,29 +354,39 @@ class CSRGeneralMatrix {
      * Sorts the column indices in each row for faster access.
      */
     void sort() {
-#pragma omp parallel for schedule(guided)
-        for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
-            if (rowIdx[i + 1] - rowIdx[i] > 1) {
-                quicksort(rowIdx[i], rowIdx[i + 1] - 1);
+#pragma omp parallel
+        {
+            std::vector<index> permutation(nCols);
+#pragma omp for schedule(guided)
+            for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
+                const index startOfRow = rowIdx[i], endOfRow = rowIdx[i + 1];
+                const count nonZerosInRow = endOfRow - startOfRow;
+                if (nonZerosInRow <= 1
+                    || std::is_sorted(columnIdx.begin() + startOfRow, columnIdx.begin() + endOfRow))
+                    continue;
```

Comment:
Why do we check whether the row is sorted first? Did you benchmark this? I would expect the `std::sort` implementation to work well on already sorted data.

Reply:
Apart from `std::sort`, the idea is to avoid the two useless `applyPermutation` calls (under the assumption that `std::is_sorted` is faster than two calls `applyPermutation` calls). However, I do not have benchmark data to confirm that.

Reply:
Okay, makes sense.